src/Compilers/Core/Portable/Collections/TemporaryArray`1.cs

```diff
@@ -214,6 +226,39 @@ public void Clear()
             }
         }
 
+        public T RemoveLast()
+        {
+            var count = this.Count;
```

Comment:
are we deliberately using this pattern instead of `this[^1]` because we want to avoid multiple reads of the Count property?

Reply:
I'm fine with either tbh. this was just clear in my head :)